jwst/lib/set_telescope_pointing.py

```diff
@@ -207,45 +212,72 @@ def update_wcs(model, default_pa_v3=0., **kwargs):
+    """Update ``S_REGION`` sky footprint information.
 
-    The S_REGION keyword is intended to store the spatial footprint of
-    an imaging observation using the VO standard STCS representation.
+    The ``S_REGION`` keyword is intended to store the spatial footprint of
+    an observation using the VO standard STCS representation.
 
     Parameters
     ----------
-    model: jwst.datamodels.DataModel
+    model : `~jwst.datamodels.DataModel`
         The model to update in-place.
+    prd_db_filepath : str
+        The filepath to the SIAF file (PRD database).
+        If None, attempt to get the path from the ``XML_DATA`` environment variable.
     """
-    wcsinfo = model.meta.wcsinfo
-    wcs = WCS(wcsinfo.instance)
-    footprint = wcs.calc_footprint(undistort=True, center=True)
-
-    # Format the region string
+    if prd_db_filepath is None:
+        try:
+            prd_db_filepath = os.path.join(os.environ['XML_DATA'], "prd.db")
+        except KeyError:
+            logger.info("Unknown path to PRD DB file: {0}".format(prd_db_filepath))
+            return
+    if not os.path.exists(prd_db_filepath):
+        logger.info("Invalid path to PRD DB file: {0}".format(prd_db_filepath))
+        return
+    aperture_name = model.meta.aperture.name
+    useafter = model.meta.observation.date
+    vertices = _get_vertices_idl(aperture_name, useafter, prd_db_filepath=prd_db_filepath)
+    vertices = list(vertices.values())[0]
+    xvert = vertices[:4]
+    yvert = vertices[4:]
+    logger.info("Vertices for aperture {0}: {1}".format(aperture_name, vertices))
+    # Execute IdealToV2V3, followed by V23ToSky
+    from ..transforms.models import IdealToV2V3, V23ToSky
+    v2_ref = model.meta.wcsinfo.v2_ref / 3600
+    v3_ref = model.meta.wcsinfo.v3_ref / 3600
+    roll_ref = model.meta.wcsinfo.roll_ref
+    ra_ref = model.meta.wcsinfo.ra_ref
+    dec_ref = model.meta.wcsinfo.dec_ref
+    vparity = model.meta.wcsinfo.vparity
+    v3yangle = model.meta.wcsinfo.v3yangle
+
+    idltov23 = IdealToV2V3(v3yangle, v2_ref, v3_ref, vparity)
+
+    angles = [-v2_ref, v3_ref, -roll_ref, -dec_ref, ra_ref]
+    axes = "zyxyz"
+    v23tosky = V23ToSky(angles, axes_order=axes)
+    idl2sky = idltov23 | v23tosky
+
+    ra_vert, dec_vert = idl2sky(xvert, yvert)
+    ind = np.argsort(ra_vert)
+    footprint = np.array([ra_vert[ind], dec_vert[ind]]).T
     s_region = (
-        'POLYGON ICRS'
-        ' {} {}'
-        ' {} {}'
-        ' {} {}'
-        ' {} {}'
-        ' {} {}'.format(
-            footprint[0][0], footprint[0][1],
-            footprint[1][0], footprint[1][1],
-            footprint[2][0], footprint[2][1],
-            footprint[3][0], footprint[3][1],
-            footprint[0][0], footprint[0][1]
-        )
-    )
-    wcsinfo.s_region = s_region
+        "POLYGON ICRS "
+        " {0} {1}"
+        " {2} {3}"
+        " {4} {5}"
+        " {6} {7}"
+        " {0} {1}".format(*footprint.flatten()))
```

Comment:
See question above as to whether this order of arguments goes counter-clockwise around the perimeter of the footprint.

Reply:
Pretty sure it's in the correct order here.